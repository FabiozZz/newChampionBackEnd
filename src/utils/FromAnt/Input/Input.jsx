import React, { useEffect, useMemo, useState } from 'react';
import { Input as Text } from 'antd';
import classes from './input.module.css';
import cn from 'classnames';

const { TextArea } = Text;
const Input = ({
	label = '',
	placeholder = 'Не заполнено',
	disabled = false,
	error,
	value,
	setValue,
	name,
	type,
	...props
}) => {
	const [input, setInput] = useState('');

	const errorInput = useMemo(() => {
		if (error) {
			return {
				boxShadow: '0 0 3px 0px red',
				borderRadius: 2,
			};
		}
		return {};
	}, [error]);
	const [oneLoad, setOneLoad] = useState(true);

	const change = e => {
		let value = e.target.value;
		setInput(value);
		if (setValue && typeof setValue === 'function') {
			setValue(e);
		}
	};
	useEffect(() => {
		if (value && oneLoad) {
			setInput(value);
			setOneLoad(false);
		}
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [value]);
	return (
		<div style={{ position: 'relative' }}>
			<span className={classes.label}>{label}</span>
			{type === 'textarea' ? (
				<TextArea
					name={name}
					{...props}
					style={errorInput}
					value={input}
					className={cn({ [classes.error]: error })}
					placeholder={placeholder}
					disabled={disabled}
					onChange={change}
				/>
			) : (
				<Text
					name={name}
					{...props}
					type={type}
					style={errorInput}
					value={input}
					className={cn({ [classes.error]: error })}
					placeholder={placeholder}
					disabled={disabled}
					onChange={change}
				/>
			)}
			{error && <span className={'form-fail'}>{error}</span>}
		</div>
	);
};

export default Input;
