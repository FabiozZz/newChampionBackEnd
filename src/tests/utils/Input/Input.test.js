import { act, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Input from 'utils/FromAnt/Input/Input';

describe('Input', function () {
	it('Input have in the document', () => {
		act(() => {
			render(<Input />);
		});
		expect(screen.getByPlaceholderText(/не заполнено/i)).toBeInTheDocument();
	});

	it('typing user in Input', () => {
		render(<Input />);
		expect(screen.queryByDisplayValue('react')).toBeNull();
		userEvent.type(screen.getByRole('textbox'), 'react');

		expect(screen.getByDisplayValue('react')).toBeInTheDocument();
	});
	it('with value and onChange', () => {
		let value = '';
		const onChange = jest.fn(e => {
			value = e.target.value;
		});
		const text = 'some text typing user';
		render(<Input value={value} setValue={onChange} />);
		expect(value).toEqual('');
		userEvent.type(screen.getByRole('textbox'), text);
		expect(screen.getByDisplayValue(text)).toBeInTheDocument();
		expect(onChange).toHaveBeenCalledTimes(text.length);
		expect(value).toEqual(text);
	});
	it('add class error with text error', () => {
		const text = 'some text typing user';
		render(<Input error={'error'} />);
		expect(screen.getByText('error')).toBeInTheDocument();
		let span = screen.getByText('error');
		expect(span).toHaveClass('form-fail');
	});
	describe('snapshots', () => {
		it('snapshots', () => {
			const input = render(<Input />);
			expect(input).toMatchSnapshot();
		});
		it('snapshots with error', () => {
			const input = render(<Input error={'error'} />);
			expect(input).toMatchSnapshot();
		});
		it('snapshots with error and value', () => {
			const input = render(<Input error={'error'} value={'some text'} />);
			expect(input).toMatchSnapshot();
		});
		it('snapshots only value', () => {
			const input = render(<Input value={'some text'} />);
			expect(input).toMatchSnapshot();
		});
	});
});
