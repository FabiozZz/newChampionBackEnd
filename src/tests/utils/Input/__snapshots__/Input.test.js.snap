// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshots snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: relative;"
      >
        <span
          class="label"
        />
        <input
          class="ant-input"
          placeholder="Не заполнено"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: relative;"
    >
      <span
        class="label"
      />
      <input
        class="ant-input"
        placeholder="Не заполнено"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input snapshots snapshots only value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: relative;"
      >
        <span
          class="label"
        />
        <input
          class="ant-input"
          placeholder="Не заполнено"
          type="text"
          value="some text"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: relative;"
    >
      <span
        class="label"
      />
      <input
        class="ant-input"
        placeholder="Не заполнено"
        type="text"
        value="some text"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input snapshots snapshots with error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: relative;"
      >
        <span
          class="label"
        />
        <input
          class="ant-input error"
          placeholder="Не заполнено"
          style="box-shadow: 0 0 3px 0px red; border-radius: 2px;"
          type="text"
          value=""
        />
        <span
          class="form-fail"
        >
          error
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: relative;"
    >
      <span
        class="label"
      />
      <input
        class="ant-input error"
        placeholder="Не заполнено"
        style="box-shadow: 0 0 3px 0px red; border-radius: 2px;"
        type="text"
        value=""
      />
      <span
        class="form-fail"
      >
        error
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Input snapshots snapshots with error and value 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="position: relative;"
      >
        <span
          class="label"
        />
        <input
          class="ant-input error"
          placeholder="Не заполнено"
          style="box-shadow: 0 0 3px 0px red; border-radius: 2px;"
          type="text"
          value="some text"
        />
        <span
          class="form-fail"
        >
          error
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="position: relative;"
    >
      <span
        class="label"
      />
      <input
        class="ant-input error"
        placeholder="Не заполнено"
        style="box-shadow: 0 0 3px 0px red; border-radius: 2px;"
        type="text"
        value="some text"
      />
      <span
        class="form-fail"
      >
        error
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
